### PR TITLE
WIP: change to the `.format()` function

### PR DESCRIPTION
**Pull request by MarkusHackspacher**
@MasterofJOKers can you please review it?

### CODE REVIEW

File: horizons/network/server.py [hunk 1]
```diff
@@ -298,61 +300,73 @@ def onreceive(self, event):
...
 			self.fatalerror(player, __("Unknown or malformed packet. Please check your game version"))
 			return
 
 		# session id check
 		if packet.sid != player.sid:
-			logging.warning("[RECEIVE] Invalid session id for player %s (%s vs %s)!" % (peer.address, packet.sid, player.sid))
-			self.fatalerror(player, __("Invalid/Unknown session")) # this will trigger ondisconnect() for cleanup
+			logging.warning(
+				"[RECEIVE] Invalid session id for player {0!s} "
+				"({1!s} vs {2!s})!"
```
Comment (MasterofJOKers) [inline horizons/network/server.py]: These don't fit in the line before? These lines look really short.

File: horizons/network/server.py [hunk 2]
```diff
@@ -298,61 +300,73 @@ def onreceive(self, event):
...
+				.format(peer.address, packet.sid, player.sid))
+			self.fatalerror(player, __("Invalid/Unknown session"))
+			# this will trigger ondisconnect() for cleanup
```
Comment (MasterofJOKers) [inline horizons/network/server.py]: If you move the comment, it should definitely go before the line that is meant, not after.

File: horizons/network/server.py [hunk 3]
```diff
@@ -298,61 +300,73 @@ def onreceive(self, event):
...
+				.format(self.capabilities['minplayers']))
 		if packet.maxplayers > self.capabilities['maxplayers']:
-			raise network.SoftNetworkException("You can't run a game with more than %d players" % (self.capabilities['maxplayers']))
+			raise network.SoftNetworkException(
+				"You can't run a game with more than {0:d} players"
+				.format(self.capabilities['maxplayers']))
 		game = Game(packet, player)
-		logging.debug("[CREATE] [%s] %s created %s" % (game.uuid, player, game))
+		logging.debug("[CREATE] [{0!s}] {1!s} created {2!s}".format(
+			game.uuid, player, game))
```
Comment (MasterofJOKers) [inline horizons/network/server.py]: You are inconsistent at where you place the `.format(` if it doesn't completely fit the line. Sometimes its on the next line and sometimes it isnt. Please make up your mind and be consistent.

File: horizons/network/server.py [hunk 4]
```diff
@@ -457,7 +471,7 @@ def leavegame(self, player):
 		if not game.is_open():
 			self.call_callbacks('terminategame', game, player)
 			return
-		logging.debug("[LEAVE] [%s] %s left %s" % (game.uuid, player, game))
+		logging.debug("[LEAVE] [{0!s}] {1!s} left {2!s}".format(game.uuid, player, game))
```
Comment (MasterofJOKers) [inline horizons/network/server.py]: How do you decide when to use `{0!s}` and when to use `{0}`? I've seen both for `game.uuid` now.

**Comment by MasterofJOKers** [comment]
Are these all occurences of not using `format()`? Are you planning on doing more in this direction? If yes (to the second question), then please add them to this pull-request and prepend a "WIP:" to the pull-request's name.

**Comment by MarkusHackspacher** [comment]
The migrated `%` string formatting are about 123 files.
https://github.com/MarkusHackspacher/unknown-horizons/commit/0fdd897e926f10def95c4d313cc7ad719fc61670

I'm planning more from this. But, not all files together, maybe ten files, than 1 month nothing, ten files and so on.

**Comment by MasterofJOKers** [comment]
ok, tell me, when you're done with the first batch. You should then open a new pull-request for the next batch when you have the time.

**Comment by MarkusHackspacher** [comment]
I'm finish with the first batch.

**Comment by MasterofJOKers** [comment]
I'll look at it as soon as possible. Thank you.

**Comment by MasterofJOKers** [comment]
I'm not very happy merging this. You are not consistent in your formatting, so I have to do a lot of changes to your changes. I've decided to use 
```python
"{0!s}". \
format(data)
```
when the line is too long. You used that, too. Additionally, you broke the line before the dot. For consistence sake, I had to decide now.

You have also mixed adding unrelated line breaks with changes to `format()` in a single commit. Although I agree, that some lines are too long, this should be separated in its own commit. You seem to know that, as you have done so for another file. I have reverted the mixed changes.

Other than that, thank you for your contribution.

**Pull request closed without merging**